substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -782,6 +877,26 @@ impl<T: Config> Pallet<T> {
 			.expect("we only map through support vector which cannot change the size; qed")
 	}
 
+	/// Adds a new set of stashes to the electable stashes.
+	///
+	/// Deduplicates stashes in place and returns an error if the bounds are exceeded. In case of
+	/// error, it returns the stashes that were not added to the storage.
+	pub(crate) fn add_electables(
+		mut stashes_iter: impl Iterator<Item = T::AccountId>,
```

Comment:
There is a lot of cloning around this code. Would be a good exercise to try and minimize them.